index.html

```diff
@@ -577,30 +598,88 @@ <h2>
         mode</a>.
       </p>
       <p>
-        Each <a>display mode</a>, except <code>browser</code>, has a
-        <dfn>fallback display mode</dfn>, which is the <a>display mode</a> that
-        the user agent can try to use if it doesn't support a particular
-        <a>display mode</a>. If the user agent does support a <a>fallback
-        display mode</a>, then it checks to see if it can use that <a>display
-        mode</a>'s <a>fallback display mode</a>. This creates a fallback chain,
-        with the <a>default display mode</a> (<code>browser</code>) being the
-        last item in the chain.
+        For more advanced usages, the {{WebAppManifest/display_override}}
+        member can be used to specify a custom fallback order.
+      </p>
+      <p>
+        The steps for determing the web app's chosen display mode is given
+        by the following algorithm. The algorithm takes a
+        [=processed manifest=] |manifest:processed manifest| and returns a
+        [=display mode=].
+        <ol>
+          <li>
+            [=list/For each=] |candidate_display_mode:DisplayModeType| of
+            |manifest|.{{WebAppManifest/display_override}}:
+            <ol>
+              <li>
+                If the user agent supports the |candidate_display_mode|, then
+                return |candidate_display_mode|.
+              </li>
+            </ol>
+          </li>
+          <li>
+            Define the |fallback_chain| for {{browser}} to be [{{browser}}]
+          </li>
+          <li>
+            Define the |fallback_chain| for <a data-link-for="DisplayModeType">
+            minimal-ui</a> to be [<a data-link-for="DisplayModeType">minimal-ui
+            </a>, {{browser}}]
+          </li>
+          <li>
+            Define the |fallback_chain| for {{standalone}} to be [
+            {{standalone}}, <a data-link-for="DisplayModeType">minimal-ui</a>,
+            {{browser}}]
+          </li>
+          <li>
+            Define the |fallback_chain| for {{fullscreen}} to be [
+            {{fullscreen}}, {{standalone}}, <a data-link-for="DisplayModeType">
+            minimal-ui</a>, {{browser}}]
+          </li>
+          <li>
+            Let |candidate_display_mode| be the the value specified by the
+            |manifest|.{{WebAppManifest/display}} property.
+          </li>
+          <li>
+            [=list/For each=] |fallback_mode:DisplayModeType| of the
+            |fallback_chain| of the |candidate_display_mode|:
+            <ol>
+              <li>
+                If the user agent supports the |fallback_mode|, then
+                return |fallback_mode|.
+              </li>
```

Comment:
The above suggestions would collapse this algorithm into just two loops: one to do display_override and one to do the fallback chain of display. I think that would look quite nice.

Arguably it's still cleaner to do my original suggestion, which is to do a single loop over "the [concatenation](https://infra.spec.whatwg.org/#string-concatenate) of *manifest*.`display_override` and the fallback chain of *manifest*.`display`".

But I will leave that up to you, since I can see how it's maybe easier to read having two loops, each with a distinct purpose.